format-common/src/main/java/io/cdap/plugin/format/plugin/AbstractFileSink.java

```diff
@@ -76,8 +96,17 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
   public void prepareRun(BatchSinkContext context) throws Exception {
     FailureCollector collector = context.getFailureCollector();
     config.validate(collector);
-    ValidatingOutputFormat validatingOutputFormat = context.newPluginInstance(FORMAT_PLUGIN_ID);
     FileFormat fileFormat = config.getFormat();
+    ValidatingOutputFormat validatingOutputFormat;
+    try {
+      validatingOutputFormat = context.newPluginInstance(fileFormat.name().toLowerCase());
+    } catch (InvalidPluginConfigException e) {
+      String errorMessage = String.format("Format '%s' cannot be used because properties %s were not provided or " +
+                                          "were invalid when the pipeline was deployed. Set the format to a " +
+                                          "different value, or re-create the pipeline with all required properties.",
+                                          fileFormat.name(), e.getMissingProperties());
```

Comment:
properties in the message should include both the missing and the invalid. Otherwise if it was from just invalid, the message will look weird.

Reply:
Added invalid properties in error msg.